modules/reactor/test/tests/meshgenerators/hexagon_mesh_trimmer/tests

```diff
@@ -0,0 +1,222 @@
+    recover = false
+  []
+  [patterned_pattern_trimming_tri]
+    type = 'Exodiff'
+    input = 'assembly_trim.i'
+    exodiff = 'patterned_pattern_trimming_tri.e'
+    exodiff_opts = '-pedantic'
+    cli_args = 'Mesh/trim/input=pattern2
+                Mesh/trim/center_trim_starting_index=4
+                Mesh/trim/center_trim_ending_index=6
+                Mesh/trim/center_trimming_section_boundary="center_section"
+                Mesh/inactive="peripheral_ring pattern_mod"
+                Mesh/hex_1/ring_block_names="center center"
+                Mesh/hex_1/ring_block_ids="15 15"
+                Mesh/hex_1/quad_center_elements=true
+                --mesh-only "patterned_pattern_trimming_tri.e"'
+    requirement ='The system shall be able to perform center trimming on a core mesh generated by double hexagonal patterning with pin centers consiting of triangular elements.'
+    recover = false
+    allow_warnings = true
+  []
+  [patterned_trimmed]
+    type = 'Exodiff'
+    input = 'patterned_trimmed.i'
+    exodiff = 'patterned_trimmed_in.e'
+    exodiff_opts = '-pedantic'
+    cli_args = '--mesh-only'
+    requirement ='The system shall be able to stitch the assembly meshes after peripheral trimming together.'
+    recover = false
+    allow_warnings = true
+  []
+  [err_incompatible]
+    type = RunException
+    input = 'err_trimmability.i'
+    cli_args = 'Mesh/inactive="hex_1 peripheral_ring"'
+    expect_err = 'The input mesh is not compatible with HexagonMeshTrimmer.'
+    requirement = "The system shall throw an error if the input mesh is not compatible."
+  []
+  [err_center_trimmible]
+    type = RunException
+    input = 'err_trimmability.i'
+    cli_args = 'Mesh/trim/input=peripheral_ring
+                Mesh/inactive="gmg"
+                Mesh/trim/center_trim_starting_index=0
+                Mesh/trim/center_trim_ending_index=6'
+    expect_err = 'The input mesh cannot be trimmed through its center.'
+    requirement = "The system shall throw an error if the input mesh cannot be center trimmed."
+  []
+  [err_unneeded_center_trim_bdry]
+    type = RunException
+    input = 'assembly_trim.i'
+    cli_args = 'Mesh/trim/center_trimming_section_boundary="center_section"'
+    expect_err = 'this input parameter is not used if center trimming is not performed.'
+    requirement = "The system shall throw an error if the input mesh is not center trimmed but the center trimming boundary is provided."
+  []
+  [err_over_center_trim]
+    type = RunException
+    input = 'assembly_trim.i'
+    cli_args = 'Mesh/trim/center_trim_starting_index=0
+                Mesh/trim/center_trim_ending_index=10'
+    expect_err = 'the remaining mesh after center trimming defined by this parameter and center_trim_ending_index must be equal or smaller than half of the input mesh.'
+    requirement = "The system shall throw an error if the center trimming yields a mesh that is over half of the input mesh."
```

Comment:
```suggestion
    requirement = "The system shall throw an error if the center trimming would yield a mesh that is over half of the input mesh, because more than half of the angular sectors are requested."
```